### PR TITLE
instance_profiling: fix detail page crash

### PR DESCRIPTION
**Pull request by baurine**
Reason: row may be undefined

image

### CODE REVIEW

File: ui/lib/apps/InstanceProfiling/pages/Detail.js [hunk 1]
```diff
@@ -134,7 +134,6 @@ export default function Page() {
         loading={isLoading && !data}
         columns={columns}
         items={data?.tasks_status || []}
-        getKey={(row) => row.id}
```
Comment (baurine) [inline ui/lib/apps/InstanceProfiling/pages/Detail.js:137]: `row => row?.id` works as well, but I prefer to just remove it, there is no warning after removing it in the console, even there are multiple rows.

**Comment by baurine** [comment]
> I don't know why it works :)

Let me check the document deeper, actually, we don't add the getKey prop in other tables.

**Comment by baurine** [comment]
According to the source code: https://github.com/microsoft/fluentui/pull/1347/files/c1fc1449ae064802e967b36022349179dbabb8b2#diff-92617c3c9fecec754c52f44e7204cec2R321 , if we don't supply the getKey prop, it will use index as the row key, but it is still weird if the row is undefined, the `getKey` shouldn't be called.

So `getKey` should be necessary when the rows need to sort, I will add it for StatementsTable and SlowQueryTable later.

**Comment by breeswish** [comment]
Maybe the core problem is why `row` can be undefined?

**Comment by baurine** [comment]
> Maybe the core problem is why `row` can be undefined?

Oh! good hint! I think it must the ShimmeredDetailsList create those empty rows, and it proved!

https://github.com/microsoft/fluentui/blob/master/packages/office-ui-fabric-react/src/components/DetailsList/ShimmeredDetailsList.base.tsx#L24

```
export class ShimmeredDetailsListBase extends React.Component<IShimmeredDetailsListProps, {}> {
  private _shimmerItems: null[];
  private _classNames: IProcessedStyleSet<IShimmeredDetailsListStyles>;

  constructor(props: IShimmeredDetailsListProps) {
    super(props);

    this._shimmerItems = props.shimmerLines ? new Array(props.shimmerLines) : new Array(SHIMMER_INITIAL_ITEMS);
  }
```

This line code doesn't judge the item whether null (they ever did, but don't know why it is removed in the master branch):

https://github.com/microsoft/fluentui/blob/master/packages/office-ui-fabric-react/src/components/List/List.tsx#L549

```
let itemKey = this.props.getKey ? this.props.getKey(item, index) : item && (item as any).key;
```

I guess the workaround is to use `row => row?.xxx`.

**Comment by breeswish** [comment]
Considering current situation, I have some ideas about improving the table, as well as resolving some `<Skeleton>` related display problems. I will open a PR for this :)

**Pull request merged**